python/fusion_engine_client/messages/configuration.py

```diff
@@ -21,6 +21,199 @@ class ConfigType(IntEnum):
     UART1_BAUD = 257
 
 
+class Direction(IntEnum):
+    ## Aligned with vehicle +x axis.
+    FORWARD = 0,
+    ## Aligned with vehicle -x axis.
+    BACKWARD = 1,
+    ## Aligned with vehicle +y axis.
+    LEFT = 2,
+    ## Aligned with vehicle -y axis.
+    RIGHT = 3,
+    ## Aligned with vehicle +z axis.
+    UP = 4,
+    ## Aligned with vehicle -z axis.
+    DOWN = 5,
+    ## Error value.
+    INVALID = 255
+
+
+class _ConfigClassGenerator:
+    """!
+    @brief Internal class for generating ConfigClass children.
+
+    These classes consist of 3 pieces:
+    - The ConfigType associated with the class.
+    - An accessor class. This is the class that's used to get/set the values for the config object.
+    - A serialization class. This is the class that (de)serializes the data for the config object.
+
+    To generate a ConfigClass:
+    - Declared as a child of NamedTuple. The NamedTuple defines the fields.
+    - Add a create_config_class decorator that takes the ConfigType and serialization Construct associated with the class.
+
+    For example:
+    ```{.py}
+        gen = _ConfigClassGenerator()
+
+        class Foo(NamedTuple):
+            x: int
+
+        FooConstruct = Struct(
+            "x" / Int32ul
+        )
+
+        @gen.create_config_class(ConfigType.BAR, FooConstruct)
+        class Bar(Foo)
```

Comment:
So to be clear, the user would only ever interact with `Bar` in this example, right? Maybe for clarity instead of foo, foo construct, and bar, do like `_FooData`, `_FooConstruct`, `Foo`